modules/document_repository/php/uploadfile.class.inc

```diff
@@ -0,0 +1,314 @@
+                    new \LORIS\Http\StringStream(
+                        json_encode($this->getUploadDocFields($id))
+                    )
+                );
+        }
+    }
+    /**
+ * Handles the document upload process
+     *
+ * @return bool
+ */
+    function uploadDocFile(): bool
+    {
+        $uploadOK       = false;
+        $editNotifier   = new \NDB_Notifier(
+            "document_repository",
+            "edit"
+        );
+        $uploadNotifier = new \NDB_Notifier(
+            "document_repository",
+            "upload"
+        );
+        $factory        = \NDB_Factory::singleton();
+        $baseURL        = $factory->settings()->getBaseURL();
+        $config         = $factory->config();
+        $base           = $config->getSetting('base');
+        $name           = \User::singleton()->getUsername();
+        $DB       = \Database::singleton();
+        $category = $_POST['category']; // required
+        $site     = $_POST['forSite']       !== '' ? $_POST['forSite'] : null;
+
+        $site       = $DB->pselectOne(
+            "SELECT CenterID FROM psc WHERE Name=:name",
+            array('name' => $site)
+        );
+        $instrument = $_POST['instrument'] !== '' ? $_POST['instrument'] : null;
+        $pscid      = $_POST['pscid']      !== '' ? $_POST['pscid'] : null;
+        $visit      = $_POST['visitLabel']      !== '' ? $_POST['visitLabel'] : null;
+        $comments   = $_POST['comments']   !== '' ? $_POST['comments'] : null;
+        $version    = $_POST['version']    !== '' ? $_POST['version'] : null;
+        $fileSize   = $_FILES["file"]["size"];
+        $fileName   = $_FILES["file"]["name"];
+        $fileType   = pathinfo($_FILES['file']['name'], PATHINFO_EXTENSION);
+        $uploadPath = "$base/modules/document_repository/user_uploads/";
+        // $category is a string representation of an ID, and so should be at
+        // least equal to zero.
+        if (intval($category) < 0) {
+            throw new \LorisException(
+                "'Category' parameter must be a positive integer."
+            );
+        }
+        // Check to see if $fullPath is writable. If not, throw an error. If it
+        // doesn't exist, create an uploads folder for the logged-in user.
+        if (!is_writable($uploadPath)) {
+            if (file_exists($uploadPath)) {
+                throw new \LorisException(
+                    "User uploads path in Document Repository is not writable."
+                );
+            }
+            mkdir($uploadPath, 0770);
+        }
+        // Copy the uploaded file to the user's upload folder if possible.
+        // Insert a record of the file into the document_repository table
+        if (!move_uploaded_file(
+            $_FILES['file']['tmp_name'],
+            $uploadPath . $fileName
+        )) {
+            throw new LorisException(
+                'ERROR: Could not upload file. Contact your administrator.'
+            );
+        } else {
```

Comment:
the else is unnecessary, since the if left the function. It can just be directly in the function body.